zvariant/src/utils.rs

```diff
@@ -316,3 +316,47 @@ where
 {
     input.get(index).ok_or(Error::OutOfBounds)
 }
+
+impl Signature<'_> {
```

Comment:
Why create another impl in a separate file? I don't see any reason for this.

Reply:
The reasoning was twofold; that it is a utility to `PartialEq` and I liked the method syntax better than the function.

Would you rather have a function in `util` or move the method to the main impl?